tests/test_device.py

```diff
@@ -14,332 +14,388 @@
 """
 Unit tests for the :mod:`pennylane` :class:`Device` class.
 """
-import unittest
-from unittest.mock import patch, PropertyMock
-import inspect
-import logging as log
-log.getLogger('defaults')
+from unittest.mock import MagicMock, Mock, PropertyMock, patch
 
-import autograd
-from autograd import numpy as np
+import pennylane as qml
+import pytest
+from pennylane import Device, DeviceError
 
-from defaults import pennylane as qml, BaseTest
-from pennylane.plugins import DefaultQubit
-from pennylane import Device
 
+@pytest.fixture(scope="function")
+def mock_device():
+    """A mock instance of the abstract Device class"""
+    with patch.multiple(Device, __abstractmethods__=set()):
+        yield Device()
 
-class DeviceTest(BaseTest):
-    """Device tests."""
-    def setUp(self):
-        self.default_devices = ['default.qubit', 'default.gaussian']
 
-        self.dev = {}
+mock_device_operations = ["PauliX", "PauliZ", "CNOT"]
 
-        for device_name in self.default_devices:
-            self.dev[device_name] = qml.device(device_name, wires=2)
 
-    def test_reset(self):
-        """Test reset works (no error is raised). Does not verify
-        that the circuit is actually reset."""
-        self.logTestName()
+@pytest.fixture(scope="function")
+def mock_device_with_operations():
+    """A mock instance of the abstract Device class with non-empty operations"""
 
-        for dev in self.dev.values():
-            dev.reset()
+    with patch.multiple(
+        Device,
+        __abstractmethods__=set(),
+        operations=PropertyMock(return_value=mock_device_operations),
```

Comment:
Since `mock_device_operations` is only used in this fixture, it makes the fixture more readable for me to have the test data within the function, i.e.
```suggestion
        operations=PropertyMock(return_value=["PauliX", "PauliZ", "CNOT"]),
```